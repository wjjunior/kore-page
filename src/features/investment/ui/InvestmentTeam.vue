<template>
  <div>
    <h2
      class="font-hanken-grotesk font-bold text-[32px] leading-[100%] tracking-[1%] text-primary-200 mb-4"
    >
      Team
    </h2>

    <div class="mb-8">
      <p
        class="text-[20px] font-normal text-black leading-[28px] tracking-[0px] mb-4"
      >
        {{ teamDescription }}
      </p>
    </div>

    <div v-if="loading" class="flex flex-wrap gap-6">
      <div
        v-for="i in 4"
        :key="i"
        class="bg-white rounded-lg border border-primary-200 p-6 w-[430px] text-center animate-pulse"
      >
        <div class="w-24 h-24 mx-auto rounded-full mb-4 bg-gray-200"></div>
        <div class="h-6 bg-gray-200 rounded mb-1 w-32 mx-auto"></div>
        <div class="h-4 bg-gray-200 rounded mb-4 w-40 mx-auto"></div>
        <div class="h-4 bg-gray-200 rounded mb-4 w-full"></div>
        <div class="h-4 bg-gray-200 rounded mb-4 w-full"></div>
        <div class="flex justify-center space-x-6">
          <div class="w-16 h-4 bg-gray-200 rounded"></div>
          <div class="w-16 h-4 bg-gray-200 rounded"></div>
          <div class="w-8 h-4 bg-gray-200 rounded"></div>
        </div>
      </div>
    </div>

    <div v-else class="flex flex-wrap gap-6">
      <div
        v-for="member in teamMembers"
        :key="member.id"
        class="bg-white rounded-lg border border-primary-200 p-6 w-[430px] text-center"
      >
        <img
          :src="member.image"
          :alt="member.name"
          class="w-24 h-24 mx-auto rounded-full mb-4 object-cover"
        />

        <h3
          class="font-bold text-[24px] leading-[100%] tracking-[1%] text-primary-200 mb-1"
        >
          {{ member.name }}
        </h3>

        <p
          class="font-medium text-[13px] leading-[20px] tracking-[0px] text-center text-gray-6 mb-4"
        >
          {{ member.position }}
        </p>

        <p
          class="font-normal text-[16px] leading-[20px] tracking-[0px] text-gray-6 mb-4 text-left"
        >
          {{ member.description }}
        </p>

        <div class="flex justify-center space-x-6 text-sm">
          <a
            v-if="member.socialLinks.linkedin"
            :href="member.socialLinks.linkedin"
            class="flex items-center space-x-1 text-gray-700"
          >
            <FontAwesomeIcon :icon="faLinkedinIn" class="text-gray-700" />
            <span class="text-gray-700">LinkedIn</span>
          </a>
          <a
            v-if="member.socialLinks.facebook"
            :href="member.socialLinks.facebook"
            class="flex items-center space-x-1 text-gray-700"
          >
            <FontAwesomeIcon :icon="faFacebookF" class="text-gray-700" />
            <span class="text-gray-700">Facebook</span>
          </a>
          <a
            v-if="member.socialLinks.twitter"
            :href="member.socialLinks.twitter"
            class="flex items-center space-x-1"
          >
            <FontAwesomeIcon :icon="faXTwitter" class="text-gray-500" />
            <span class="text-gray-500">x</span>
          </a>
        </div>
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
import {
  faFacebookF,
  faLinkedinIn,
  faXTwitter,
} from "@fortawesome/free-brands-svg-icons";
import type { TeamMember } from "@/shared/lib/types";

defineOptions({
  name: "InvestmentTeam",
});

interface InvestmentTeamProps {
  teamMembers: TeamMember[];
  teamDescription: string;
  loading?: boolean;
}

withDefaults(defineProps<InvestmentTeamProps>(), {
  loading: false,
});
</script>
