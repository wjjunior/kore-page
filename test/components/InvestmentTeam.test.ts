import { describe, it, expect, vi, beforeEach } from "vitest";
import { mount } from "@vue/test-utils";
import { createPinia, setActivePinia } from "pinia";
import InvestmentTeam from "@/features/investment/ui/InvestmentTeam.vue";
import { mockTeamMembers } from "../mocks";

const mountInvestmentTeam = (props: {
  teamMembers: typeof mockTeamMembers;
  teamDescription: string;
  loading?: boolean;
}) =>
  mount(InvestmentTeam, {
    props,
  });

describe("InvestmentTeam", () => {
  beforeEach(() => {
    setActivePinia(createPinia());
    vi.clearAllMocks();
  });

  it("renders the section title and description", () => {
    const wrapper = mountInvestmentTeam({
      teamMembers: mockTeamMembers,
      teamDescription: "Meet our experienced team",
    });
    expect(wrapper.text()).toContain("Team");
    expect(wrapper.text()).toContain("Meet our experienced team");
  });

  it("renders all team members with name, position, and description", () => {
    const wrapper = mountInvestmentTeam({
      teamMembers: mockTeamMembers,
      teamDescription: "Team",
    });

    for (const member of mockTeamMembers) {
      expect(wrapper.text()).toContain(member.name);
      expect(wrapper.text()).toContain(member.position);
      expect(wrapper.text()).toContain(member.description);
    }
  });

  it("renders team member images with correct attributes", () => {
    const wrapper = mountInvestmentTeam({
      teamMembers: mockTeamMembers,
      teamDescription: "Team",
    });

    const images = wrapper.findAll("img");
    expect(images).toHaveLength(mockTeamMembers.length);
    mockTeamMembers.forEach((member, index) => {
      const img = images[index];
      expect(img?.attributes("src")).toBe(member.image);
      expect(img?.attributes("alt")).toBe(member.name);
      expect(img?.attributes("loading")).toBe("lazy");
      expect(img?.attributes("format")).toBe("webp");
    });
  });

  it("renders LinkedIn, Facebook, and Twitter links when available", () => {
    const wrapper = mountInvestmentTeam({
      teamMembers: mockTeamMembers,
      teamDescription: "Team",
    });

    expect(wrapper.findAll('a[href*="linkedin.com"]')).toHaveLength(2);
    expect(wrapper.findAll('a[href*="facebook.com"]')).toHaveLength(1);
    expect(wrapper.findAll('a[href*="twitter.com"]')).toHaveLength(1);
  });

  it("does not render social links if not provided", () => {
    const wrapper = mountInvestmentTeam({
      teamMembers: [
        {
          id: 3,
          name: "No Social",
          position: "Analyst",
          image: "/img.jpg",
          description: "Test desc",
          socialLinks: {},
        },
      ],
      teamDescription: "Team",
    });

    expect(wrapper.findAll("a")).toHaveLength(0);
  });

  it("renders loading skeleton when loading is true", () => {
    const wrapper = mountInvestmentTeam({
      teamMembers: mockTeamMembers,
      teamDescription: "Loading test",
      loading: true,
    });

    expect(wrapper.findAll(".animate-pulse")).toHaveLength(4);
    expect(wrapper.text()).not.toContain("John Doe");
  });

  it("renders empty state gracefully with no team members", () => {
    const wrapper = mountInvestmentTeam({
      teamMembers: [],
      teamDescription: "No one yet",
    });

    expect(wrapper.text()).toContain("Team");
    expect(wrapper.text()).toContain("No one yet");
    expect(wrapper.findAll("img")).toHaveLength(0);
  });

  it("matches snapshot for loaded state", () => {
    const wrapper = mountInvestmentTeam({
      teamMembers: mockTeamMembers,
      teamDescription: "Snapshot test",
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot for loading state", () => {
    const wrapper = mountInvestmentTeam({
      teamMembers: mockTeamMembers,
      teamDescription: "Snapshot loading",
      loading: true,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot for empty state", () => {
    const wrapper = mountInvestmentTeam({
      teamMembers: [],
      teamDescription: "Snapshot empty",
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
