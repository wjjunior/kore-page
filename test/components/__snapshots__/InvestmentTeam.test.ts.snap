// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InvestmentTeam > matches snapshot for empty state 1`] = `
"<div>
  <h2 class="font-hanken-grotesk font-bold text-2xl md:text-3xl lg:text-[30px] xl:text-[32px] leading-[100%] tracking-[1%] text-primary-200 mb-4"> Team </h2>
  <div class="mb-8">
    <p class="text-lg md:text-xl lg:text-[19px] xl:text-[20px] font-normal text-black leading-[28px] tracking-[0px] mb-4">Snapshot empty</p>
  </div>
  <div class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-2 xl:grid-cols-2 gap-4 lg:gap-5 xl:gap-6"></div>
</div>"
`;

exports[`InvestmentTeam > matches snapshot for loaded state 1`] = `
"<div>
  <h2 class="font-hanken-grotesk font-bold text-2xl md:text-3xl lg:text-[30px] xl:text-[32px] leading-[100%] tracking-[1%] text-primary-200 mb-4"> Team </h2>
  <div class="mb-8">
    <p class="text-lg md:text-xl lg:text-[19px] xl:text-[20px] font-normal text-black leading-[28px] tracking-[0px] mb-4">Snapshot test</p>
  </div>
  <div class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-2 xl:grid-cols-2 gap-4 lg:gap-5 xl:gap-6">
    <div class="bg-white rounded-lg border border-primary-200 p-4 lg:p-5 xl:p-6 w-full max-w-[380px] lg:max-w-[400px] xl:max-w-[430px] text-center mx-auto"><img src="/images/john-doe.jpg" alt="John Doe" class="w-24 h-24 mx-auto rounded-full mb-4 object-cover" loading="lazy" format="webp">
      <h3 class="font-bold text-[24px] leading-[100%] tracking-[1%] text-primary-200 mb-1">John Doe</h3>
      <p class="font-medium text-[13px] leading-[20px] tracking-[0px] text-center text-gray-6 mb-4">CEO</p>
      <p class="font-normal text-[16px] leading-[20px] tracking-[0px] text-gray-6 mb-4 text-left">Experienced CEO with 15+ years in fintech</p>
      <div class="flex justify-center flex-wrap gap-4 text-sm"><a href="https://linkedin.com/in/johndoe" target="_blank" rel="noopener noreferrer" aria-label="LinkedIn profile" class="flex items-center space-x-1 text-gray-700"><svg class="svg-inline--fa fa-linkedin-in text-gray-700" data-prefix="fab" data-icon="linkedin-in" role="img" viewBox="0 0 448 512" aria-hidden="true">
            <path class="" fill="currentColor" d="M100.3 448l-92.9 0 0-299.1 92.9 0 0 299.1zM53.8 108.1C24.1 108.1 0 83.5 0 53.8 0 39.5 5.7 25.9 15.8 15.8s23.8-15.8 38-15.8 27.9 5.7 38 15.8 15.8 23.8 15.8 38c0 29.7-24.1 54.3-53.8 54.3zM447.9 448l-92.7 0 0-145.6c0-34.7-.7-79.2-48.3-79.2-48.3 0-55.7 37.7-55.7 76.7l0 148.1-92.8 0 0-299.1 89.1 0 0 40.8 1.3 0c12.4-23.5 42.7-48.3 87.9-48.3 94 0 111.3 61.9 111.3 142.3l0 164.3-.1 0z"></path>
          </svg><span class="text-gray-700 hidden sm:inline">LinkedIn</span></a><a href="https://facebook.com/johndoe" target="_blank" rel="noopener noreferrer" aria-label="Facebook profile" class="flex items-center space-x-1 text-gray-700"><svg class="svg-inline--fa fa-facebook-f text-gray-700" data-prefix="fab" data-icon="facebook-f" role="img" viewBox="0 0 320 512" aria-hidden="true">
            <path class="" fill="currentColor" d="M80 299.3l0 212.7 116 0 0-212.7 86.5 0 18-97.8-104.5 0 0-34.6c0-51.7 20.3-71.5 72.7-71.5 16.3 0 29.4 .4 37 1.2l0-88.7C291.4 4 256.4 0 236.2 0 129.3 0 80 50.5 80 159.4l0 42.1-66 0 0 97.8 66 0z"></path>
          </svg><span class="text-gray-700 hidden sm:inline">Facebook</span></a><a href="https://twitter.com/johndoe" target="_blank" rel="noopener noreferrer" aria-label="Twitter profile" class="flex items-center space-x-1"><svg class="svg-inline--fa fa-x-twitter text-gray-500" data-prefix="fab" data-icon="x-twitter" role="img" viewBox="0 0 448 512" aria-hidden="true">
            <path class="" fill="currentColor" d="M357.2 48L427.8 48 273.6 224.2 455 464 313 464 201.7 318.6 74.5 464 3.8 464 168.7 275.5-5.2 48 140.4 48 240.9 180.9 357.2 48zM332.4 421.8l39.1 0-252.4-333.8-42 0 255.3 333.8z"></path>
          </svg><span class="text-gray-500 hidden sm:inline">x</span></a></div>
    </div>
    <div class="bg-white rounded-lg border border-primary-200 p-4 lg:p-5 xl:p-6 w-full max-w-[380px] lg:max-w-[400px] xl:max-w-[430px] text-center mx-auto"><img src="/images/jane-smith.jpg" alt="Jane Smith" class="w-24 h-24 mx-auto rounded-full mb-4 object-cover" loading="lazy" format="webp">
      <h3 class="font-bold text-[24px] leading-[100%] tracking-[1%] text-primary-200 mb-1">Jane Smith</h3>
      <p class="font-medium text-[13px] leading-[20px] tracking-[0px] text-center text-gray-6 mb-4">CTO</p>
      <p class="font-normal text-[16px] leading-[20px] tracking-[0px] text-gray-6 mb-4 text-left">Tech leader</p>
      <div class="flex justify-center flex-wrap gap-4 text-sm"><a href="https://linkedin.com/in/janesmith" target="_blank" rel="noopener noreferrer" aria-label="LinkedIn profile" class="flex items-center space-x-1 text-gray-700"><svg class="svg-inline--fa fa-linkedin-in text-gray-700" data-prefix="fab" data-icon="linkedin-in" role="img" viewBox="0 0 448 512" aria-hidden="true">
            <path class="" fill="currentColor" d="M100.3 448l-92.9 0 0-299.1 92.9 0 0 299.1zM53.8 108.1C24.1 108.1 0 83.5 0 53.8 0 39.5 5.7 25.9 15.8 15.8s23.8-15.8 38-15.8 27.9 5.7 38 15.8 15.8 23.8 15.8 38c0 29.7-24.1 54.3-53.8 54.3zM447.9 448l-92.7 0 0-145.6c0-34.7-.7-79.2-48.3-79.2-48.3 0-55.7 37.7-55.7 76.7l0 148.1-92.8 0 0-299.1 89.1 0 0 40.8 1.3 0c12.4-23.5 42.7-48.3 87.9-48.3 94 0 111.3 61.9 111.3 142.3l0 164.3-.1 0z"></path>
          </svg><span class="text-gray-700 hidden sm:inline">LinkedIn</span></a>
        <!--v-if-->
        <!--v-if-->
      </div>
    </div>
  </div>
</div>"
`;

exports[`InvestmentTeam > matches snapshot for loading state 1`] = `
"<div>
  <h2 class="font-hanken-grotesk font-bold text-2xl md:text-3xl lg:text-[30px] xl:text-[32px] leading-[100%] tracking-[1%] text-primary-200 mb-4"> Team </h2>
  <div class="mb-8">
    <p class="text-lg md:text-xl lg:text-[19px] xl:text-[20px] font-normal text-black leading-[28px] tracking-[0px] mb-4">Snapshot loading</p>
  </div>
  <div class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-2 xl:grid-cols-2 gap-4 lg:gap-5 xl:gap-6">
    <div class="bg-white rounded-lg border border-primary-200 p-4 lg:p-5 xl:p-6 w-full max-w-[380px] lg:max-w-[400px] xl:max-w-[430px] text-center animate-pulse mx-auto">
      <div class="w-24 h-24 mx-auto rounded-full mb-4 bg-gray-200"></div>
      <div class="h-6 bg-gray-200 rounded mb-1 w-32 mx-auto"></div>
      <div class="h-4 bg-gray-200 rounded mb-4 w-40 mx-auto"></div>
      <div class="h-4 bg-gray-200 rounded mb-4 w-full"></div>
      <div class="h-4 bg-gray-200 rounded mb-4 w-full"></div>
      <div class="flex justify-center space-x-6">
        <div class="w-16 h-4 bg-gray-200 rounded"></div>
        <div class="w-16 h-4 bg-gray-200 rounded"></div>
        <div class="w-8 h-4 bg-gray-200 rounded"></div>
      </div>
    </div>
    <div class="bg-white rounded-lg border border-primary-200 p-4 lg:p-5 xl:p-6 w-full max-w-[380px] lg:max-w-[400px] xl:max-w-[430px] text-center animate-pulse mx-auto">
      <div class="w-24 h-24 mx-auto rounded-full mb-4 bg-gray-200"></div>
      <div class="h-6 bg-gray-200 rounded mb-1 w-32 mx-auto"></div>
      <div class="h-4 bg-gray-200 rounded mb-4 w-40 mx-auto"></div>
      <div class="h-4 bg-gray-200 rounded mb-4 w-full"></div>
      <div class="h-4 bg-gray-200 rounded mb-4 w-full"></div>
      <div class="flex justify-center space-x-6">
        <div class="w-16 h-4 bg-gray-200 rounded"></div>
        <div class="w-16 h-4 bg-gray-200 rounded"></div>
        <div class="w-8 h-4 bg-gray-200 rounded"></div>
      </div>
    </div>
    <div class="bg-white rounded-lg border border-primary-200 p-4 lg:p-5 xl:p-6 w-full max-w-[380px] lg:max-w-[400px] xl:max-w-[430px] text-center animate-pulse mx-auto">
      <div class="w-24 h-24 mx-auto rounded-full mb-4 bg-gray-200"></div>
      <div class="h-6 bg-gray-200 rounded mb-1 w-32 mx-auto"></div>
      <div class="h-4 bg-gray-200 rounded mb-4 w-40 mx-auto"></div>
      <div class="h-4 bg-gray-200 rounded mb-4 w-full"></div>
      <div class="h-4 bg-gray-200 rounded mb-4 w-full"></div>
      <div class="flex justify-center space-x-6">
        <div class="w-16 h-4 bg-gray-200 rounded"></div>
        <div class="w-16 h-4 bg-gray-200 rounded"></div>
        <div class="w-8 h-4 bg-gray-200 rounded"></div>
      </div>
    </div>
    <div class="bg-white rounded-lg border border-primary-200 p-4 lg:p-5 xl:p-6 w-full max-w-[380px] lg:max-w-[400px] xl:max-w-[430px] text-center animate-pulse mx-auto">
      <div class="w-24 h-24 mx-auto rounded-full mb-4 bg-gray-200"></div>
      <div class="h-6 bg-gray-200 rounded mb-1 w-32 mx-auto"></div>
      <div class="h-4 bg-gray-200 rounded mb-4 w-40 mx-auto"></div>
      <div class="h-4 bg-gray-200 rounded mb-4 w-full"></div>
      <div class="h-4 bg-gray-200 rounded mb-4 w-full"></div>
      <div class="flex justify-center space-x-6">
        <div class="w-16 h-4 bg-gray-200 rounded"></div>
        <div class="w-16 h-4 bg-gray-200 rounded"></div>
        <div class="w-8 h-4 bg-gray-200 rounded"></div>
      </div>
    </div>
  </div>
</div>"
`;
